### PR TITLE
[ONNX] Fix MatMul op for vec @ tensor multiplication

### PR DESCRIPTION
**Pull request by arogowie-intel**
Fix MatMul implementation for cases where left hand side of operation is a vector.

### CODE REVIEW

**Comment by arogowie-intel** [comment]
TESTNOW

**Comment by arogowie-intel** [comment]
TEST NOW

File: src/ngraph/frontend/onnx_import/op/matmul.cpp [hunk 1]
```diff
@@ -71,24 +76,30 @@ namespace ngraph
                     std::size_t left_rank{left->get_shape().size()};
                     std::size_t right_rank{right->get_shape().size()};
 
-                    // First (easy) case:
-                    // Multiply two tensors where one of them or both has rank lower equal 2.
-                    // This is already internally handled by Ngraph Dot operator.
-                    if (left_rank <= 2 || right_rank <= 2)
+                    ASSERT_VALID_ARGUMENT(node, (left_rank != 0 && right_rank != 0))
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/matmul.cpp]: Log a warning instead.

File: src/ngraph/frontend/onnx_import/op/matmul.cpp [hunk 2]
```diff
@@ -71,24 +77,35 @@ namespace ngraph
...
 
-                    // First (easy) case:
-                    // Multiply two tensors where one of them or both has rank lower equal 2.
-                    // This is already internally handled by Ngraph Dot operator.
-                    if (left_rank <= 2 || right_rank <= 2)
+                    if (left_rank == 0 || right_rank == 0)
+                    {
+                        NGRAPH_WARN
+                            << (node) << " "
+                            << "ONNX standard doesn't allow scalar operands, however Ngraph "
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/matmul.cpp]: Bad spelling of `nGraph`

File: test/onnx_import.cpp [hunk 3]
```diff
@@ -1346,3 +1346,19 @@ TEST(onnx, model_custom_op_default_domain)
     Outputs outputs{execute(function, inputs, "INTERPRETER")};
     EXPECT_TRUE(test::all_close_f(expected_outputs.front(), outputs.front()));
 }
+
+TEST(onnx, model_matmul_vec_ten3D)
```
Comment (postrational) [inline test/onnx_import.cpp]: `3D` -> `3d`

File: test/onnx_import.cpp [hunk 4]
```diff
@@ -1346,3 +1346,19 @@ TEST(onnx, model_custom_op_default_domain)
...
+{
+    auto function = onnx_import::import_onnx_function(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/matmul_vec_ten3D.onnx"));
```
Comment (postrational) [inline test/onnx_import.cpp]: `3D` -> `3d`

File: src/ngraph/frontend/onnx_import/op/matmul.cpp [hunk 5]
```diff
@@ -97,11 +114,20 @@ namespace ngraph
...
                         left = reshape::collapse(left, 0, left_shape.size() - 3);
+                    }
+                    if (right_shape.size() > 3)
+                    {
                         right = reshape::collapse(right, 0, right_shape.size() - 3);
                     }
 
                     // Perform multiple small dot products
                     std::size_t groups = left->get_shape().at(0);
+                    // If we haven't make broadcasting earlier this means that one of inputs is vector,
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/matmul.cpp]: Please reword:
`If we haven't broadcast earlier this means that one of the inputs is a vector,`

**Pull request merged**